### PR TITLE
[macOS] Enable WEBCONTENT_GPU_SANDBOX_EXTENSIONS_BLOCKING

### PR DESCRIPTION
**Pull request by pvollan**
#### 37058f49d69567fee5bf978118d9b1fdc3e31f1b
<pre>
[macOS] Enable WEBCONTENT_GPU_SANDBOX_EXTENSIONS_BLOCKING
<a href="https://bugs.webkit.org/show_bug.cgi?id=265076">https://bugs.webkit.org/show_bug.cgi?id=265076</a>
<a href="https://rdar.apple.com/114834841">rdar://114834841</a>

Reviewed by Brent Fulgham.

This feature flag can be enabled on newer OS versions.

* Source/WTF/wtf/PlatformEnableCocoa.h:
* Source/WebKit/WebProcess/com.apple.WebProcess.sb.in:

Canonical link: <a href="https://commits.webkit.org/274941@main">https://commits.webkit.org/274941@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/8a6c19806e38d876f933c22964242dee6786b60c

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/40448 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/19460 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/42826 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/42996 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/36537 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/22414 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/16789 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/33558 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/41022 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/16414 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/34872 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/14142 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/14209 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/35844 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/44272 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/33901 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/36675 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/36167 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/39911 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/40074 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/15263 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/12506 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/38207 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/16882 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/47084 "Built successfully") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/16932 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/9689 "Passed tests") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/5364 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/16526 "Built successfully") | | |

### CODE REVIEW

File: Source/WebKit/WebProcess/com.apple.WebProcess.sb.in [hunk 1]
```diff
@@ -43,13 +43,23 @@
...
-    "AGPM"
-    "AppleAPFSContainer"
-    "AppleM2ScalerCSCDriver"
-    "AppleVideoToolboxParavirtualizationDriver"
-    "IOAccelerator"))
+(define (webcontent_gpu_sandbox_extensions_blocking)
+#if ENABLE(WEBCONTENT_GPU_SANDBOX_EXTENSIONS_BLOCKING)
+    (equal? (param "CPU") "arm64")
+#else
+    ;; false
```
Comment (brentfulgham) [inline Source/WebKit/WebProcess/com.apple.WebProcess.sb.in]: Does `#f` not work for this?

**Review by brentfulgham — COMMENTED** [review]
This seems to be failing everywhere. Does it need to be rebased?

File: Source/WebKit/WebProcess/com.apple.WebProcess.sb.in [hunk 2]
```diff
@@ -2309,19 +2016,16 @@
...
-        (iokit-user-client-class
-            "RootDomainUserClient"
-        )
-    )
-)
+        (iokit-user-client-class "RootDomainUserClient")))
+
+(if (webcontent_gpu_sandbox_extensions_blocking)
+(deny iokit-open-service (with telemetry-backtrace))
+(deny iokit-open-user-client (with telemetry-backtrace)))
```
Comment (brentfulgham) [inline Source/WebKit/WebProcess/com.apple.WebProcess.sb.in]: Nit: Can you fix the indentation on these two lines?

**Review by brentfulgham — APPROVED** [review]
Please correct the indentation issue I mentioned, and see if using the `#f` primitive would work instead of the inequality test. Otherwise, r=me

**Comment by pvollan** [comment]
> Please correct the indentation issue I mentioned, and see if using the `#f` primitive would work instead of the inequality test. Otherwise, r=me

Ah, good point! I wonder if `#f` will be affected by preprocessing? I will try it.

Thanks for reviewing!

**Pull request merged**